modules/core/shared/src/main/scala/coursier/core/Definitions.scala

```diff
@@ -62,6 +63,12 @@ object ModuleName {
     s"${organization.value}:${name.value}"
 
   override final lazy val hashCode = tuple.hashCode()
+
+  def copy(
```

Comment:
I'm wary of the binary compatibility story of `copy` methods… Couldn't we add more fine grained methods, like
```scala
def update(
  org: Organization,
  name: Name
): Module = …
```
with no default values in particular.

It may also be interesting to generate a full `copy` method from data-class, but marked as `private`. That way, we would call that private `copy` from additional methods like `update`.

Reply:
as long this is not an autogenerated copy method then it is a binary compatible one, right? 

I can try to make a PR to data-class, but that might take while...

Reply:
I was thinking, we could just mark this `private[core]`, and be done with it (no need to consider binary compatibility then):
```suggestion
  private[core] def copy(
```

Reply:
Sorry I forgot about this PR...

Reply:
@alexarchambault ready for review/merge...